docs/how-to/rocks/add-internal-user-in-rocks.rst

```diff
@@ -0,0 +1,55 @@
+.. _add_internal_user_in_rocks:
+
+How to add an internal user to a rock
+*************************************
+
+Rockcraft allows you to use ``run-user`` to specify the user you want to run
+the rock's service(s) with. Without using that field in your rockcraft.yaml,
+the service(s) will be run by the root user. The downside, however, of using
+``run-user`` is that it only supports one user called ``_daemon_`` with a
+UID/GID of 84792. For most daemon services that's sufficient, but other
+services require that the running user should have a specific username and/or
+UID.
+
+In order to add a new internal user or group to a rock, two packages (or their
+respective slices) are needed:
+- ``base-files``: to create the base dirs such as ``/root`` and ``/home``,
+- ``base-passwd``: to produce basic ``/etc/passwd`` and ``/etc/group`` files.
+
+Invoking the ``useradd`` and ``groupadd`` commands can take place in the
+:ref:`build <lifecycle>` step, by appending the commands to the build flow.
+However, the changes made by those commands normally apply to the base system,
+not the build system. Hence, ``$CRAFT_PART_INSTALL`` should be passed as the
+root directory to those two commands.
+
+The following *rockcraft.yaml* illustrates how to use ``useradd`` command in
+``override-build`` field inside a part.
+
+.. literalinclude:: ../code/internal-user/rockcraft.yaml
+    :language: yaml
+    :start-after: [docs:rock-parts]
+    :end-before: [docs:rock-parts-end]
+
+Furthermore, let's add a service that tests the running user by executing
```

Comment:
This is a very fast context switch.
Also, it's doing more than that. It's **running a service as that user**

Reply:
What do you mean? the context here is about adding a user, then testing that it exists. If we keep it simple, there shouldn't be a step in between.

The service runs whoami to prove that pebble services can be run with the created user as well. There are other form of tests can be done, but this one was like hitting two birds with one stone (or rock) imo.

Reply:
I mean that the text jumps from the actual "adding a new user" to "running a Pebble service that user", without much context. There's no bridge between the two sections, so it's a very abrupt context switch.

> The service runs whoami to prove that pebble services can be run with the created user as well. There are other form of tests can be done, but this one was like hitting two birds with one stone (or rock) IMO.

Yes, but again, it's lacking the details of what's happening. The important bit is not the `whoami` but the fact that you can enforce the execution of a service to be owned by your user, **regardless of the OCI user**

Reply:
I changed both the flow and the contents of that part. Furthermore, the service was changed to a python http service.